### PR TITLE
Update installer

### LINKED ISSUE #289: Updating installer
We need to update our installer to our changed situation regarding libs and dependencies. I've created a new advanced installer project from scratch. Please check it out on my new branch (https://github.com/flokuep/Vocaluxe/tree/installer).

We need to think about

- [x] Requirements we want to check before installation
- [x] Install some depedencies automatically (.NET? DirectX?)
- [x] Do we want to provide a single installer for x86 and x64 or do we want seperate ones? (currently both; ~270 MB)

What needs to be done

- [x] Check if INSTALLER target works correctly
- [x] Find a nice process to build and maybe upload installer; x64 exe needs to be in x64-Folder, x86 exe needs to be in x86 folder; we have to build installer target and get all nuget dependencies... (checkout existing batch scripts for this or advanced installer build events)
- [ ] Check how advanced installer auto update work
- [x] Check why x64 installer doesn't work (currently build installer, that just installs x86)
- [x] Do git pull in batch-script if Files folder aready exists

Builds for testing:
- https://cloud.floriankuepper.de/index.php/s/QxOfR4gUKdviYGc
- https://cloud.floriankuepper.de/index.php/s/5DauSMoSNhtVyM0

### PR DESCRIPTION
**Pull request by flokuep**
This updates our installer, closes #289

- Updates installer to our current files
- Updates advanced installer version
- Separate x86 and x46 installer to decrease file size
- Automatically install .NET 4.0 and vc2010 runtime if not installed
- Ask during uninstallation if data folders should be deleted or not
- Fixes some issues with "Installer" build target
- New batch script for easier building installer

For testing:
- x86: https://cloud.floriankuepper.de/index.php/s/QxOfR4gUKdviYGc
- x64: https://cloud.floriankuepper.de/index.php/s/5DauSMoSNhtVyM0

### CODE REVIEW

**Comment by lukeIam** [comment]
From my dependency tests:
I uninstalled VC2012 an it's still working (if the check in `CProgrammHelper.cs` is disabled)
at least I could start Vocaluxe and play a song with video - no audio, but that was a vm problem (no audio device there)

So system dependencies are:
- VC2010
- DirectX 9c
- .NET 4.0

-> Summary: everything als expected...

File: Vocaluxe/CProgrammHelper.cs [hunk 1]
```diff
@@ -116,13 +116,20 @@ private static void _EnsureDataFolderExists()
         public static bool CheckRequirements()
         {
 #if WIN
-            if (!_IsVC2012Installed())
+            if (!_IsVC2010Installed())
```
Comment (lukeIam) [inline Vocaluxe/CProgrammHelper.cs]: @flokuep 
I removed the VC2012 check (seems not to be needed anymore) 
I added VC2010 check (you got DirectX missing error, if DirectX was installed but VC2010 not)

Comment (flokuep) [reply Vocaluxe/CProgrammHelper.cs]: Okay. Thanks.

**Comment by BenKmann** [comment]
I tried the x64 Setup and it worked fine. Installed correctly and game starts.
I don't know if it has to, but the setup didn't create a "config.xml" File. That one is missing.
The Game also didn't create one (in the Readme it is still referenced).

**Comment by flokuep** [comment]
> Or am I overlooking something?

@lukeIam No, you're right. I've changed it.

> I tried the x64 Setup and it worked fine. Installed correctly and game starts. I don't know if it has to, but the setup didn't create a "config.xml" File. That one is missing. The Game also didn't create one (in the Readme it is still referenced).

@BenKmann Thanks for your feedback. You can find config.xml in Documents/Vocaluxe. We need to put it there, because maybe we haven't write permission on install folder during running vocaluxe. I've updated readme about this, just need to merge it ;)

**Pull request merged**